schema.go

```diff
@@ -842,6 +842,21 @@ func makeNodeOf(t reflect.Type, name string, tag []string) Node {
 		}
 	})
 
+	// Special case: an "optional" struct tag on a slice applies to the
+	// individual items, not the overall list. The least messy way to
+	// deal with this is at this level, instead of passing down optional
+	// information into the nodeOf function, and then passing back whether an
+	// optional tag was applied.
+	if node == nil && t.Kind() == reflect.Slice {
+		isUint8 := t.Elem().Kind() == reflect.Uint8
+		// TODO: does 'optional string' apply to each byte?
```

Comment:
String values are represented as BYTE_ARRAY columns, not as repeated bytes, in which case the optional tag would apply to the string itself, not each byte.